From 53782cfc7b580892d5ed0585ef47c0492416892d Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Wed, 20 Nov 2013 21:52:37 -0500
Subject: [PATCH 29/29] Refresh XMPP authentication in process_jobs()

... if the username and password is known, to avoid problems with
the SASL token timing out.
---
 cloudprint/cloudprint.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index f9a4a3a..ab0e70a 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -383,6 +383,15 @@ def process_jobs(cups_connection, cpp, printers):
             LOGGER.error('ERROR: Could not Connect to Cloud Service. Will Try again in %d Seconds' % FAIL_RETRY)
             time.sleep(FAIL_RETRY)
 
+        if cpp.username and not xmpp_conn.is_connected():
+            LOGGER.debug('Refreshing authentication')
+            cpp.set_auth('')
+            try:
+                cpp.get_auth()
+                xmpp_auth = file(cpp.xmpp_auth_path).read()
+            except:
+                LOGGER.debug('Error refreshing authentication')
+
 
 def usage():
     print sys.argv[0] + ' [-d][-l][-h] [-p pid_file] [-a account_file]'
-- 
1.8.4.4

